### DISCUSSION TITLE
Sending request to FastAPI endpoint with form data OAuth2PasswordRequestForm results in Error 422

### DISCUSSION THREAD

**Discussion by DucNgn** [fastapi/fastapi#6187] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post("/login/access-token", response_model=schemas.Token)
def login_access_token(
    db: Session = Depends(deps.get_db), form_data: OAuth2PasswordRequestForm = Depends()
) -> Any:
    """
    OAuth2 compatible token login, get an access token for future requests
    """
    user = crud.user.authenticate(
        db, email=form_data.username, password=form_data.password
    )
    if not user:
        raise HTTPException(status_code=400, detail="Incorrect email or password")
    elif not crud.user.is_active(user):
        raise HTTPException(status_code=400, detail="Inactive user")
    access_token_expires = timedelta(minutes=settings.ACCESS_TOKEN_EXPIRE_MINUTES)
    return {
        "access_token": security.create_access_token(
            user.id, expires_delta=access_token_expires
        ),
        "token_type": "bearer",
    }
```

### Description

I'm making an endpoint to validate the request from the front end and return back to the client a login token. The FastAPI code is pretty straight forward like above (I took the code from [the example full-stack app](https://github.com/tiangolo/full-stack-fastapi-postgresql/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/backend/app/app/api/api_v1/endpoints/login.py#L22-L43))

In the front end, I have this function to call the `login/access-token` endpoint:

```javascript
  login(email: string, password: string): HttpReponse {
    console.log(email)
    console.log(password)
    const currentHeaders = new Headers();
    currentHeaders.append('accept', 'application/json');
    currentHeaders.append('Content-Type', 'application/x-www-form-urlencoded');
    currentHeaders.append("Access-Control-Allow-Origin", "*");

    const formData = new FormData();
    formData.append('username', email);
    formData.append('password', password);

    const requestOptions = {
      url: this.baseUrl,
      method: Verb.POST,
      mode: this.mode,
      headers: currentHeaders,
      data: formData,
    };

    return fetch(this.baseUrl + Endpoints.login, requestOptions)
      .then(response => {
        console.log(response);
        const headers = response.headers;
        this.headers['Authorization'] = headers.get('Authorization');
        return response;
      });
  }

```

However, when I try to make the request from the front end, I'm receiving `422 Unprocessable Entity`, with a response as followed:

```
{"detail":[{"loc":["body","username"],"msg":"field required","type":"value_error.missing"},{"loc":["body","password"],"msg":"field required","type":"value_error.missing"}]}
```

So I'm suspecting it has to be something from my request that makes the data invalid to be received in the back-end.

If I make the request using the swagger UI or curl from my terminal, the endpoint works as expected:
The curl command below worked fine and I receive a token from it.
```
 curl -X 'POST' \
  'http://localhost:8000/api/v1/login/access-token' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/x-www-form-urlencoded' \
  -d 'grant_type=&username=user%40user.user&password=password&scope=&client_id=&client_secret='

```

### Operating System

macOS

### FastAPI Version

0.86.0

### Python Version

3.10.8

**Comment by tiangolo** [comment]
Hey there!

Please add a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) that I can copy-paste to replicate it.

**Comment by JarroVGIT** [comment]
If your endpoint works but does not play nicely with your front end code, it’s not really a FastAPI issue I would say. Maybe Stackoverflow is a better place to ask this?

**Comment by DucNgn** [comment]
Thanks for the advice, I'll be closing this issue and asking the question on StackOverFlow instead.

**Comment by tiangolo** [comment]
Thanks for the help here @JarroVGIT ! 👏  🙇 

Thanks for reporting back and closing the issue @DucNgn  👍

**Comment by fabthegreat** [comment]
Hi @DucNgn , have you found the solution to your issue? Have the same here... Thanks
